### PR TITLE
Icons: fix having to import from /dist

### PR DESCRIPTION
**Pull request by christianvuerings**
# Why is this a breaking change?

Previous imports including `/dist` will no longer work, example change when this lands:

```diff
- import ChevronRight from "@cambly/syntax-icons/dist/ChevronRight";
+ import ChevronRight from "@cambly/syntax-icons/ChevronRight";
```

# Why is this no longer an issue in Cambly-Frontend

Since we now use a new module resolution: https://github.com/Cambly/Cambly-Frontend/pull/9878

# Notes

* Tested the change in Cambly-Frontend with a Syntax icons beta release
  * https://github.com/Cambly/Cambly-Frontend/pull/9931 
  * https://www.npmjs.com/package/@cambly/syntax-icons/v/0.3.0-beta.0?activeTab=code

### CODE REVIEW

File: packages/syntax-icons/src/packageExports.server.test.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import packageJSON from "../package.json";
+import { readdir } from "node:fs/promises";
+import path from "node:path";
+
+describe("package.json: exports", () => {
+  it("contains correct exports", async () => {
```
Comment (christianvuerings) [inline packages/syntax-icons/src/packageExports.server.test.ts:6]: @somethiiing added a test to ensure we use have the correct `exports` field

**Review by zprong — APPROVED** [review]
LG thanks for following up on this!

File: packages/syntax-core/src/Badge/Badge.stories.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { type StoryObj, type Meta } from "@storybook/react";
 import Badge from "./Badge";
 import RepeatIcon from "@mui/icons-material/Repeat";
-import Refresh from "../../../syntax-icons/src/Refresh";
+import Refresh from "../../../syntax-icons/src/icons/Refresh";
```
[the remark this answers is not in the document]
Comment (somethiiing) [reply packages/syntax-core/src/Badge/Badge.stories.tsx:4]: oh man, i wanted to do this before, but i remember you wanted to keep it flatter in core so i kept it this way

**Review by somethiiing — APPROVED** [review]
thanks for doing this, i was going to do it yesterday, but got pulled into some meetings. great job though!

**Pull request merged**